### ISSUE TITLE
test fails due to add_foreign_key

### ISSUE THREAD

**Issue by johan-smits** [rails/rails#37062]
### Steps to reproduce

1. `rm db/test.sqlite3`
2. `rails db:migrate RAILS_ENV=test`
3. `rails test`

### Expected behavior
Run tests normal

### Actual behavior
The database is dropped and newly created, this adds changes to the schema and it adds foreign_keys:
```ruby
add_foreign_key "relations", "relation_institutes", column: "institute_id"
```

When the test runs it fails with this message
```bash
Migrations are pending. To resolve this issue, run:

        rails db:migrate RAILS_ENV=test
```

### System configuration
**Rails version**: `6.0.0`

**Ruby version**: `2.5.1p57`

### More info

When I remove the added `add_foreign_key` from the schema tests run normal. The project was created from `5.2` and upgraded to `6.0.0`.

**Comment by y-yagi** [comment]
Can you please provide a sample application that reproduces the issue?

**Comment by johan-smits** [comment]
Sorry for the delay, here is the example project: https://github.com/johan-smits/rails-add-foreign-key
There are 2 commits, one after `bundle update` to the 6.0 version. And one after `rails migrate`.

**Comment by sinsoku** [comment]
@johan-smits Can you tell me how to reproduce the issue in the example project?

I executed the following commands but did not see the "Migrations are pending." message.

```console
$ git clone https://github.com/johan-smits/rails-add-foreign-key.git
$ cd rails-add-foreign-key
$ bundle install
$ rails db:migrate RAILS_ENV=test
$ rails test
Run options: --seed 24422

# Running:



Finished in 0.000589s, 0.0000 runs/s, 0.0000 assertions/s.
0 runs, 0 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by johan-smits** [comment]
@sinsoku sure I can.

1. Check out commit: johan-smits/rails-add-foreign-key@37e9e4f45d9a9f6011339f797411c7918414027a
2. `$ rm db/test.sqlite3`
3. `$ rails db:migrate RAILS_ENV=test`
4. now you can see changes to the schema as comitted in commit johan-smits/rails-add-foreign-key@b41a5532db49ceb7f3a79c73126a984d846f9703

**Comment by sinsoku** [comment]
@johan-smits Thank you for the comment on the reproduction procedure.

I have see the changes of schema.rb, but I think that it is the intended behavior because there is `foreign_key: true` in the `20190314135126_create_location_categories.rb`.
Also I couldn't check the "Migrations are pending" message when I ran the test after this steps.

I checked steps to reproduce in both the local environemnt and Docker, but I cannot check the issue.
Are there any other necessary steps?

ref: https://github.com/sinsoku/rails-add-foreign-key/commit/46e6ac8ad2f66e13c80707b66ff39d7d4cf5c61b
